src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+
+    private String name = "Rectangle";
+    private double firstLeg, secondLeg;
+    private String color;
+
+    public Rectangle(double firstLeg, double secondLeg, String color) {
```

Comment:
Creating a figure, don't pass expressions in the constructor. It's better to create local variables in the 'FigureSupplier' class and pass them to the constructor for better code readability.